code/game/objects/structures/safe.dm

```diff
@@ -210,10 +278,27 @@ FLOOR SAFES
 		if(istype(I, /obj/item/clothing/accessory/stethoscope))
 			to_chat(user, "<span class='warning'>Hold [I] in one of your hands while you manipulate the dial!</span>")
 			return
+		else if(istype(I, /obj/item/thermal_drill))
+			if(drill)
+				to_chat(user, "There is already a drill attached!")
+			else if(do_after(user, 2 SECONDS, target = src))
+				if(!user.drop_item())
+					to_chat(user, "<span class='warning'>[I] is stuck to your hand, you cannot put it in the safe!</span>")
+					return
+				I.loc = src

```

Comment:
It could be better to use `forceMove()` for this but a maintainer's opinion would be good.